src/contracts/swivel.ts

```diff
@@ -197,9 +246,7 @@ export class Swivel {
         const amounts = a.map(amount => BigNumber.from(amount));
         const signatures = s.map(signature => utils.splitSignature(signature));
 
-        const options = await optimizeGas(t, this.contract, 'initiate', orders, amounts, signatures);
-
-        return await this.contract.functions.initiate(orders, amounts, signatures, options) as TransactionResponse;
+        return await this.executor(this.contract, 'initiate', [orders, amounts, signatures], t, true);
```

Comment:
it looks strange to have `this.executor(this.contract, ...`

as `executor` is on `this` and `contract` is on `this` - doesn't that mean that `contract` would be visible to `executor` without being passed?

Reply:
`executor` might look like a method of the `Swivel` class, but if you look at the constructor, it's not:

`executor` is a class property which holds a method, but that method is passed into the `Swivel` class at construction time. It's not scoped and doesn't have access to the `Swivel` class' prototype or instance. The reason that I pass that method is so that I can swap it out during tests. The executor will perform transaction validation and gas optimization, which will fail during a test when using bogus contract addresses.